packages/events-api/package.json

```diff
@@ -36,40 +36,39 @@
   },
   "scripts": {
     "prepare": "npm run build",
-    "build": "babel src -d dist --source-maps both",
-    "lint": "eslint src test",
-    "test": "npm run build && nyc --reporter=html mocha test/**/*.js",
-    "coverage": "nyc report --reporter=text-lcov > coverage.lcov && codecov -F eventsapi"
+    "build": "npm run build:clean && tsc",
+    "build:clean": "shx rm -rf ./dist ./coverage ./.nyc_output",
+    "lint": "tslint --project .",
+    "test": "nyc mocha --config .mocharc.json src/*.spec.js test/integration/*.js",
+    "coverage": "codecov -F eventsapi --root=$PWD"
   },
   "dependencies": {
     "debug": "^2.6.1",
-    "lodash.isstring": "^4.0.1",
     "raw-body": "^2.3.3",
     "tsscmp": "^1.0.6",
     "yargs": "^6.6.0"
   },
   "devDependencies": {
-    "babel-cli": "^6.18.0",
-    "babel-eslint": "^7.1.1",
-    "babel-preset-es2015": "^6.18.0",
-    "babel-preset-es2016": "^6.16.0",
-    "chai": "^4.1.2",
+    "@types/debug": "^4.1.4",
```

Comment:
Only those types that show up in the exposed API (`.d.ts` files) need to be under `"dependencies"`. I checked the `dist/` dirs of each of the two packages and both only have a `<reference />` to `@types/node`. I should change the dependencies of (only) these two to only include `@types/node` in `"dependencies"` and other dependent types in `"devDependencies"`

Reply:
While I agree that we can get away with not putting any packages whose API is not exposed into `"devDependencies"`, what problems could arise from putting them all in `"dependencies"`?

What I'd like to avoid is a complex heuristic for judging each `@types/*` package individually in order to correctly spread them across the two distinctions. I'm okay with slightly less optimized `npm install` time and `node_modules` size if the win is that we verifiably know that the type definitions are available when needed. In lieu of an integration test that can do that verification, I believe it's easier to just put them all in `"dependencies"` so that it's harder to make a mistake.

Reply:
🤔 Hum. Let me share my process for **solidly** and **verifiably** determining whether a given `@types/` package should be in `"dependencies"` or `"devDependencies"`:

1. `npm run build`
2. check built `.d.ts` files for the `<reference types="xyz" />` comments at the top of the file
  a. packages that are `<reference />`'d should be in `"dependencies"`
  b. all others should be in `"devDependencies"`

Creating a script (e.g. after building) or integration test to do that feels just as simple. Pass any `package/*/dist/**/*.d.ts` file through something that extracts the `<reference />` comments (i.e. programmatically compile as a TypeScript program or just match against the regex `^// <reference types="([^"]+)" />$`), then make sure all and only all referenced types packages are in `"dependencies"` (for its respective `package.json`).

Reply:
That seems like quite a bit of added complexity. What gain are we expecting in return? I can see some benefit from quicker downloads of dependencies, but this is very very small.

I believe its this same cost/benefit analysis that has led the TypeScript project maintainers to recommend just putting them all in `"dependencies"` (which cannot be wrong) instead of splitting the dependencies across the two (which can be wrong unless we write some specific verification tooling, and continue to maintain it).

Reply:
Can we agree that the idea to separate `@types/*` dependencies according to whether or not they are re-exported via any of the package's own interfaces can become a separate proposal (which could apply to each of the packages)?

If so, I'd like to move forward by keeping all `@types/*` dependencies in `"dependencies"`, and then carry out that discussion in a separate issue. I think that will make it easier for others to weigh in, too (this discussion is buried in the code review of one PR). If we reach a different conclusion there, we can make the updates across all the packages at once.

Reply:
Sounds valid to me! ➡️ #830